src/configuration.js

```diff
@@ -0,0 +1,252 @@
+import { ENV_VARS } from 'apify-shared/consts';
+import { join } from 'path';
+import { ApifyStorageLocal } from '@apify/storage-local';
+import * as ApifyClient from 'apify-client';
+import log from './utils_log';
+
+/**
+ * `Configuration` is a value object holding the SDK configuration. We can use it in two ways:
+ *
+ * 1. When using `Apify` class, we can get the instance configuration via `sdk.config`
+ *   ```js
+ *   const { Apify } = require('apify-js');
+ *
+ *   const sdk = new Apify({ token: '123' });
+ *   console.log(sdk.config.get('token')); // '123'
+ *   ```
+ * 2. To get the global configuration (singleton instance). It will respect the environment variables.
+ *   ```js
+ *   console.log(Configuration.getGlobalConfig().get('token')); // returns the token from APIFY_TOKEN env var
+ *   ```
+ *
+ * ## Supported Configuration Options
+ *
+ * > All env vars are also accessible with `APIFY_` prefix (e.g. `APIFY_TOKEN`)
+ *
+ * Key | Environment Variable | Default Value
+ * ---|---|---
+ * `token` | `TOKEN` | -
+ * `localStorageDir` | `LOCAL_STORAGE_DIR` | `'./apify_storage'`
+ * `localStorageEnableWalMode` | `LOCAL_STORAGE_ENABLE_WAL_MODE` | `true`
+ * `defaultDatasetId` | `DEFAULT_DATASET_ID` | `'default'`
+ * `defaultKeyValueStoreId` | `DEFAULT_KEY_VALUE_STORE_ID` | `'default'`
+ * `defaultRequestQueueId` | `DEFAULT_REQUEST_QUEUE_ID` | `'default'`
+ * `metamorphAfterSleepMillis` | `METAMORPH_AFTER_SLEEP_MILLIS` | `300e3`
+ * `persistStateIntervalMillis` | `PERSIST_STATE_INTERVAL_MILLIS` | `60e3`
+ * `actorEventsWsUrl` | `ACTOR_EVENTS_WS_URL` | -
+ * `inputKey` | `INPUT_KEY` | `'INPUT'`
+ * `actorId` | `ACTOR_ID` | -
+ * `apiBaseUrl` | `API_BASE_URL` | `'https://api.apify.com/v2'`
+ * `isAtHome` | `IS_AT_HOME` | -
+ * `actorRunId` | `ACTOR_RUN_ID` | -
+ * `actorTaskId` | `ACTOR_TASK_ID` | -
+ * `containerPort` | `CONTAINER_PORT` | `4321`
+ * `containerUrl` | `CONTAINER_URL` | `'http://localhost:4321'`
+ * `userId` | `USER_ID` | -
+ * `proxyHostname` | `PROXY_HOSTNAME` | `'proxy.apify.com'`
+ * `proxyPassword` | `PROXY_PASSWORD` | -
+ * `proxyStatusUrl` | `PROXY_STATUS_URL` | `'http://proxy.apify.com'`
+ * `proxyPort` | `PROXY_PORT` | `8000`
+ * `maxOpenedStorages` | `MAX_OPENED_STORAGES` | `1000`
+ *
+ * ## Not Supported environment variables
+ *
+ * - `MEMORY_MBYTES`
+ * - `HEADLESS`
+ * - `XVFB`
+ * - `CHROME_EXECUTABLE_PATH`
+ */
+export class Configuration {
+    // all env vars also supports `APIFY_` prefix
```

Comment:
I don't think it worked without the prefix because I remember that sometimes I defined `TOKEN` instead of `APIFY_TOKEN` and it took me a few minutes to figure out what's wrong. I also tested it right now. With that said, there might be some place or some env var for which it worked, but that would be a bug.

Reply:
Yes, never worked that way, just misread the code, already fixed.